galasa-maven-plugin/src/test/java/dev/galasa/maven/plugin/auth/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,238 @@
+
+        String expectedClientId = "asdasdasads";
+        String expectedRefreshToken = "valid-token";
+        String expectedJwt = "my-jwt";
+
+        MockHttpClient mockHttpClient = new MockHttpClient() {
+            @Override
+            public HttpResponse execute(HttpUriRequest request) throws IOException, ClientProtocolException {
+                // Note that the code under test has asked for execution of an Http request...
+                super.incrementRequestsProcessedCount();
+
+                assertThat(request.getURI().toString()).isEqualTo("https://mock-service.com/auth");
+
+                HttpEntity entity = ((HttpEntityEnclosingRequest) request).getEntity();
+                String requestBodyString = EntityUtils.toString(entity);
+
+                Gson gson = new GsonFactory().getGson();
+                AuthRequestPayload payload = gson.fromJson(requestBodyString, AuthRequestPayload.class);
+
+                assertThat(payload.client_id).as("Client id field in request to auth endpoint is bad.").isEqualTo(expectedClientId);
+                assertThat(payload.code).isNull();
+                assertThat(payload.refresh_token).as("refresh token field in request to auth endpoint is bad.").isEqualTo(expectedRefreshToken);
+                assertThat(payload.secret).isNull();
+
+                // The request looks OK.
+
+                // Formulate a mock response...
+                StatusLine statusLine = new BasicStatusLine(HttpVersion.HTTP_1_1, HttpStatus.SC_BAD_REQUEST, "");
+                BasicHttpResponse response = new BasicHttpResponse(statusLine);
+
+                AuthError authError = new AuthError();
+                authError.error_code = 99;
+                authError.error_message = "The galasa token was bad";
+
+                String responseBodyString = gson.toJson(authError);
+                StringEntity responsePayloadEntity = new StringEntity(responseBodyString, ContentType.APPLICATION_JSON);
+                response.setEntity(responsePayloadEntity);
+
+                return response;
+            }
+        };
+
+        AuthenticationService service = new AuthenticationService(new URL("https://mock-service.com"), expectedRefreshToken+":"+expectedClientId, mockHttpClient);
+
+        Throwable t = catchThrowable( ()-> { service.getJWT(); });
+
+        assertThat(t).isNotNull().isInstanceOf(AuthenticationException.class);
+        AuthenticationException ex = (AuthenticationException)t;
+        assertThat(ex).hasMessageContaining("The galasa token was bad");
+
+    }
+
+
+    // This test was helpful to show that the AuthenticationService code should work against a real server.
+    // To use it, you need to plug-in your own token.
+    // @Test
+    // public void testCanTargetARealEcosystem() throws Exception {
+    //     String refreshToken = "xxx";
+    //     String clientId = "xxx";
+    //     String apiServerUrlString = "https://galasa-ecosystem1.galasa.dev/api";
```

Comment:
I think apiServerUrlString should point to a dummy url here i.e. `https://my.server.url/api`

Reply:
Done.